File: 20190209-java-phone-numbers/src/test/java/com/katas/PhoneListTest.java
package com.katas;

import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class PhoneListTest {
    @Test
    public void should_be_consistent_when_has_one_phone_number() {
        var phoneList = new PhoneList(List.of("911"));

        assertThat(phoneList.isConsistent()).isTrue();
    }

    @Test
    public void should_not_be_consistent_when_a_number_is_prefix_of_another() {
        var phoneList = new PhoneList(List.of("911", "91176"));

        assertThat(phoneList.isConsistent()).isFalse();
    }

    @Test
    public void should_be_consistent_when_any_number_is_prefix_of_another() {
        var phoneList = new PhoneList(List.of("911", "976", "12110"));

        assertThat(phoneList.isConsistent()).isTrue();
    }
}

File: 20181125-kotlin-phone-numbers/src/test/kotlin/com/katas/PhoneListTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class PhoneListTest {

    @Test
    fun `should be consistent when is empty`() {
        val phoneList = PhoneList()

        val isConsistent = phoneList.isConsistent()

        assertThat(isConsistent).isTrue()
    }

    @Test
    fun `should be consistent when has a single phone number`() {
        val phoneList = PhoneList.of("91125426")

        val isConsistent = phoneList.isConsistent()

        assertThat(isConsistent).isTrue()
    }

    @Test
    fun `should not be consistent when a number if prefix of another one`() {
        val phoneList = PhoneList.of("91125426", "911")

        val isConsistent = phoneList.isConsistent()

        assertThat(isConsistent).isFalse()
    }
}